pkg/operator/options/options_validation.go

```diff
@@ -39,6 +41,20 @@ func (o Options) Validate() error {
 	)
 }
 
+func (o Options) validateVNETGUID() error {
+	if !o.isAzureCNIWithOverlay() {
```

Comment:
This seems suspicious to me.
If it's not overlay, but they pass an invalid vnet GUID, we allow it?

Should we parse the vnetGUID if it's supplied always, and then also enforce it's set when overlay is set?

Reply:
We only use the vnet guid if overlay is set, so if it's malformed, I wouldn't want to crash the customers' application if they don't need the value. 

Thoughts?

Reply:
Ok, I agree with this approach. If we are going to do it this way, lets use uuid.Validate() instead as it just returns a bool along with an error. 

```go
	if o.VnetGUID != "" && uuid.Validate(o.VnetGUID) != nil {
		return fmt.Errorf("vnet-guid %s is malformed", o.VnetGUID)
	}
	if o.isAzureCNIWithOverlay() && o.VnetGUID == "" {
		return fmt.Errorf("vnet-guid cannot be empty for AzureCNI clusters with networkPluginMode overlay")
	}
```

Reply:
Ok modified to use this approach, so I will be resolving this comment.